ros2bag/ros2bag/verb/record.py

```diff
@@ -90,6 +90,13 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             '--qos-profile-overrides-path', type=FileType('r'),
             help='Path to a yaml file defining overrides of the QoS profile for specific topics.'
         )
+        parser.add_argument(
+            '--resilient-storage-writing', action='store_true',
```

Comment:
@emersonknapp that would likely be better in the long run, if there are sensible presets other than optimized/safer. Is this a blocker for this PR? The primary goal of this PR is to change the default optimization for the storage.

Reply:
@emersonknapp I am fine with either, please let me know if I should put it into this PR or a separate one.

Reply:
I would prefer to do this change now - the `resiliant_storage_writing` value is the sort of thing that seems likely to linger even though it is an overly narrow API. Calling is a `preset_config` probably also does a good job of clarifying that the `storage-config-file` overrides these values.

If you will do the followup change, though, I won't block this PR.

Reply:
@emersonknapp I made an attempt to implement it, but there are some drawbacks to each solution I came up with. There is currently no ros2_storage interface exposed to the python (except for some structs) and, since this API requires plugin-specific return value, one would need to instantiate storage to return these profiles. Which would be the only case when this is done, but the storage remains not-open and is discarded after the call. 

Another solution would be to have a separate object hierarchy (and a factory) just for the profiles.

Yet another would be to simply keep a dictionary of plugin name - preset profiles at the top level, and return based on which plugin is meant to be used (as in the argument).

Each solution has drawbacks and I would love to discuss that with you. Did you have a particular design in mind? We could also set up a call if that would make things easier, let me know.